pyroute2/dhcp/leases.py

```diff
@@ -93,22 +94,45 @@ def ip(self) -> str:
         return self.ack['yiaddr']
 
     @property
-    def subnet_mask(self) -> str:
+    def subnet_mask(self) -> int | None:
         '''The subnet mask assigned to the client.'''
-        return self.ack['options']['subnet_mask']
+        mask = self.ack['options'].get('subnet_mask')
+        if mask is None:
+            return None
+        if isinstance(mask, int) or mask.isdigit():
+            return int(mask)
+        return dqn2int(mask)
+
+    @property
+    def routers(self) -> list[str]:
+        return self.ack['options'].get('router', [])
+
+    @property
+    def default_gateway(self) -> str | None:
+        '''The default gateway for this interface.
+
+        As mentioned by the RFC, the first router is the most prioritary.
+        '''
+        return self.routers[0] if self.routers is not None else None
```

Comment:
Is it possible to get here an instance of a type other than `(tuple, list, None)`?

Reply:
I think the condition is wrong anyway; in cases where the `routers` option wasn't provided, `self.routers` will return `[]`, and `self.routers[0]` will raise `IndexError` because the code checks for `None` explicitly.